### PR TITLE
Makecode init vs import fix

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**

### CODE REVIEW

File: src/components/Editor.tsx [hunk 1]
```diff
@@ -18,21 +17,13 @@ const Editor = forwardRef<MakeCodeFrameDriver, EditorProps>(function Editor(
...
   ref
 ) {
-  const logging = useLogging();
-  const { project, editorCallbacks } = useProject();
-  const initialProjects = useCallback(() => {
-    logging.log(
-      `[MakeCode] Initialising with header ID: ${project.header?.id}`
-    );
-    return Promise.resolve([project]);
-  }, [logging, project]);
```
Comment (microbit-matt-hillsdon) [inline src/components/Editor.tsx:28]: I just moved this into useProjects as it seemed odd that only one aspect of the MakeCode integration wasn't in that hook.

**Comment by microbit-robert** [comment]
This appears to have broken loading / draging and dropping a hex (into our app as opposed to inside MakeCode which I've not tested yet).

**Comment by microbit-grace** [comment]
> have

Dragging to MakeCode works. I think it's to do with the added "isEditorOpen" if in the editorChange store action

**Comment by microbit-matt-hillsdon** [comment]
I think ac8a0e195d4606578f0bd8c41b0a8f5d0b180f79 addresses these. Let's test carefully on Monday.

Another PR could add timeouts/error handling to the places that wait for MakeCode to be ready as they'll be the bits that fail if it doesn't load.

**Review by microbit-grace — COMMENTED** [review]
LGTM, tried dnd on MakeCode and tool, opening session, and the import page. Not sure what Rob thinks

**Comment by microbit-robert** [comment]
We don't get a toast notification for dragging and dropping a JSON file, but I think this is fine for now. Data samples don't really feel like a "project", and maybe this will get addressed further with the data replace / append option.

**Review by microbit-robert — APPROVED** [review]
LGTM. Every route seems to work well. Added a comment about JSON file imports.

**Pull request merged**